File: coding/22nd March 2025/Reverse Integer.py
#Reverse Integer
class Solution:
    def reverse(self, x: int) -> int:
        n = abs(x)
        s = str(n)
        rev = ""
        for ch in s:
            rev = ch + rev
        res = int(rev)
        if x < 0 :  
            res = -res
        if 2147483648 > res > -2147483649:
            return res 
        return 0

File: coding/21st September 2025/Reverse Integer.java
//Reverse Integer
class Solution 
{
    public int reverse(int x) 
    {
        int temp = Math.abs(x);
        StringBuilder sb = new StringBuilder(Integer.toString(temp));
        sb = sb.reverse();
        String str = sb.toString();
        try
        {
           int result = Integer.parseInt(str);
            if(x < 0)
                return -result;
            return result; 
        }
        catch(NumberFormatException e)
        {
            return 0;
        }
    }
}
